octavia-cli/octavia_cli/headers.py

```diff
@@ -0,0 +1,114 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+from dataclasses import dataclass
+from typing import Optional, List
+
+import click
+import yaml
+
+from .apply.yaml_loaders import EnvVarLoader
+
+
+API_HEADERS_YAML_FILE_INSTRUCTION = """
+headers:
+  - name: Authorization
+    value: Basic dXNlcjpwYXNzd29yZA==
+  - name: Content-Type
+    value: application/json
+""".strip()
+
+
+@dataclass
+class ApplicationHeader:
+    name: Optional[str]
+    value: Optional[str]
+
+    @property
+    def is_valid(self):
+        return all([self.name, self.value])
+
+    def clean(self):
+        return ApplicationHeader(
+            name=self.name.strip(),
+            value=self.value.strip()
+        )
+
+
+@dataclass
+class ApiHeadersResult:
+    raw_length: int
+    validated_length: int
+    valid_application_headers: List[ApplicationHeader]
+
+    def is_valid(self):
+        return self.raw_length == self.validated_length
```

Comment:
Could you explain why you need this object to exist instead of using a pure list of `ApplicationHeader`?

Reply:
Removed